examples/data-objects/codemirror/src/index.ts

```diff
@@ -15,41 +14,41 @@ import {
     buildRuntimeRequestHandler,
 } from "@fluidframework/request-handler";
 import { defaultRouteRequestHandler } from "@fluidframework/aqueduct";
-
+import { RuntimeFactoryHelper } from "@fluidframework/runtime-utils";
 import { fluidExport as smde } from "./codemirror";
 
-class CodeMirrorFactory implements IRuntimeFactory {
-    public get IRuntimeFactory() { return this; }
+const defaultComponentId = "default";
+const defaultComponent = "@fluid-example/smde";
+
+class CodeMirrorFactory extends RuntimeFactoryHelper {
+    public async instantiateFirstTime(runtime: ContainerRuntime): Promise<void> {
+        await runtime.createRootDataStore(defaultComponent, defaultComponentId);
+    }
 
-    public async instantiateRuntime(context: IContainerContext): Promise<IRuntime> {
+    public async preInitialize(
+        context: IContainerContext,
+        existing: boolean,
+    ): Promise<IRuntime & ContainerRuntime> {
         const registry = new Map<string, Promise<IFluidDataStoreFactory>>([
-            ["@fluid-example/smde", Promise.resolve(smde)],
+            [defaultComponent, Promise.resolve(smde)],
         ]);
 
-        const defaultComponentId = "default";
-        const defaultComponent = "@fluid-example/smde";
-
-        const runtime = await ContainerRuntime.load(
+        const runtime: ContainerRuntime = await ContainerRuntime.load(
             context,
             registry,
             buildRuntimeRequestHandler(
                 defaultRouteRequestHandler(defaultComponentId),
-                innerRequestHandler));
+                innerRequestHandler,
+            ),
+            undefined, // runtimeOptions
+            undefined, // containerScope
+            existing,
```

Comment:
Should existing arg move up in the order, given that it's not really optional any more, and the other two args are optional (I believe)

Reply:
Hmm.. `existing` is still optional (in this change). Let me check if I can make it mandatory

Reply:
it maybe fine to keep it optional, but given that eventually it should not, it makes sense to move the order of args.

Reply:
I agree but it will still break the build as there is current usage as `containerScope` is often not specified while `runtimeOptions` often is. So the effect will be similar to making this mandatory..

Are you implying that we should move it after `runtimeOptions` and before `containerScope`?

Reply:
I'd move it before runtimeOptions. This is with assumption that it will become non-optional at some point, and thus it needs to be before any optional args - doing it in this order now will reduce churn later.

Reply:
Getting a ton of back-compat related failures in the end to end tests if I mess with these parameters.. :(

Reply:
ContainerRuntime.load() API is not part of back compat promise, so it would be great to understand why you are getting these errors. IRuntime / IRuntimeFactory are, but their implementations and usage of COntainerRuntime are at the same layer, so there should not be any concerns here

Reply:
https://github.com/microsoft/FluidFramework/compare/main...andre4i:stateful-iruntimefactory-existing-mandatory?expand=1 is the branch with that change.

I'll return with a couple of errors after the build is done as I since lost the previous test run.

Reply:
My suspicion is that somewhere in the tests the class is injected -> I didn't catch the `load` invocation with find-replace in VSCode to adjust it to the new signature.